### ISSUE TITLE
Container exited with error code 1

### ISSUE THREAD

**Issue by Xeudoron** [SAIC-iSmart-API/saic-python-mqtt-gateway#118]
Hello,

for some reason I am unable to run this container. Tried an addon in multiple HA instances and even on bare metal  (Debian 12 with compose) and every time outputs the same code. I've tried wrong username or password for SAIC API and MQTT broker, which displays different for me readable error log. Until 2 days ago, this addon was working like a charm and upgrade did not help. I have tried older versions of the image but that didn't help either. Here are my logs and setup. What could be wrong?
Thank you for any help.

```
saic-mqtt-gateway  | 2023-12-11 14:42:56,496 [ DEBUG  ] Sending CONNECT (u1, p1, wr1, wq0, wf1, c1, k60) client_id=b'saic-python-mqtt-gateway' - paho.mqtt.client
saic-mqtt-gateway  | 2023-12-11 14:42:56,570 [ DEBUG  ] Received CONNACK (0, 0) - paho.mqtt.client
saic-mqtt-gateway  | 2023-12-11 14:42:56,570 [  INFO  ] Connected to MQTT broker - mqtt_publisher
saic-mqtt-gateway  | 2023-12-11 14:42:56,570 [ DEBUG  ] Sending SUBSCRIBE (d0, m1) [(b'saic/<myusername@example.com>/vehicles/+/+/+/set', 0)] - paho.mqtt.client
saic-mqtt-gateway  | 2023-12-11 14:42:56,570 [ DEBUG  ] Sending SUBSCRIBE (d0, m2) [(b'saic/<myusername@example.com>/vehicles/+/+/+/+/set', 0)] - paho.mqtt.client
saic-mqtt-gateway  | 2023-12-11 14:42:56,570 [ DEBUG  ] Sending SUBSCRIBE (d0, m3) [(b'saic/<myusername@example.com>/vehicles/+/refresh/mode/set', 0)] - paho.mqtt.client
saic-mqtt-gateway  | 2023-12-11 14:42:56,570 [ DEBUG  ] Sending SUBSCRIBE (d0, m4) [(b'saic/<myusername@example.com>/vehicles/+/refresh/period/+/set', 0)] - paho.mqtt.client
saic-mqtt-gateway  | 2023-12-11 14:42:56,570 [ DEBUG  ] Sending PUBLISH (d0, q0, r1, m5), 'b'saic/_internal/lwt'', ... (6 bytes) - paho.mqtt.client
saic-mqtt-gateway  | 2023-12-11 14:42:56,571 [ DEBUG  ] Received SUBACK - paho.mqtt.client
saic-mqtt-gateway  | 2023-12-11 14:42:56,573 [  INFO  ] Scheduler started - apscheduler.scheduler
saic-mqtt-gateway  | 2023-12-11 14:42:56,573 [ DEBUG  ] Sending PUBLISH (d0, q0, r1, m6), 'b'saic/501_513/json/request'', ... (587 bytes) - paho.mqtt.client
saic-mqtt-gateway  | 2023-12-11 14:42:56,573 [ DEBUG  ] Sending PUBLISH (d0, q0, r1, m7), 'b'saic/501_513/raw/request'', ... (219 bytes) - paho.mqtt.client
saic-mqtt-gateway  | 2023-12-11 14:42:56,573 [ DEBUG  ] Received SUBACK - paho.mqtt.client
saic-mqtt-gateway  | 2023-12-11 14:42:56,573 [ DEBUG  ] Received SUBACK - paho.mqtt.client
saic-mqtt-gateway  | 2023-12-11 14:42:56,574 [ DEBUG  ] Received SUBACK - paho.mqtt.client
saic-mqtt-gateway  | 2023-12-11 14:42:57,136 [ DEBUG  ] Sending PUBLISH (d0, q0, r1, m8), 'b'saic/501_513/raw/response'', ... (2035 bytes) - paho.mqtt.client
saic-mqtt-gateway  | 2023-12-11 14:42:57,136 [ DEBUG  ] Message length in bytes: 1015.0 - saic_ismart_client.common_model
saic-mqtt-gateway  | 2023-12-11 14:42:57,136 [ DEBUG  ] Protocol version: 17 - saic_ismart_client.common_model
saic-mqtt-gateway  | 2023-12-11 14:42:57,136 [ DEBUG  ] Dispatcher message length: 129 - saic_ismart_client.common_model
saic-mqtt-gateway  | 2023-12-11 14:42:57,136 [ DEBUG  ] Message size without header: 1011.0 - saic_ismart_client.common_model
saic-mqtt-gateway  | 2023-12-11 14:42:57,137 [ DEBUG  ] Dispatcher message bytes: 125 - saic_ismart_client.common_model
saic-mqtt-gateway  | 2023-12-11 14:42:57,139 [WARNING ] Executing <Task finished name='Task-1' coro=<MqttGateway.run() done, defined at /usr/src/app/./mqtt_gateway.py:297> exception=DecodeError('Expected enumeration index 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 or 16, but got 17.') created at /usr/local/lib/python3.11/asyncio/runners.py:100> took 0.568 seconds - asyncio
saic-mqtt-gateway  | 2023-12-11 14:42:57,139 [ DEBUG  ] Looking for jobs to run - apscheduler.scheduler
saic-mqtt-gateway  | 2023-12-11 14:42:57,140 [ DEBUG  ] No jobs; waiting until a job is added - apscheduler.scheduler
saic-mqtt-gateway  | Traceback (most recent call last):
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/site-packages/asn1tools/codecs/per.py", line 1437, in decode_root
saic-mqtt-gateway  |     data = self.root_index_to_data[index]
saic-mqtt-gateway  |            ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^
saic-mqtt-gateway  | KeyError: 17
saic-mqtt-gateway  | 
saic-mqtt-gateway  | During handling of the above exception, another exception occurred:
saic-mqtt-gateway  | 
saic-mqtt-gateway  | Traceback (most recent call last):
saic-mqtt-gateway  |   File "/usr/src/app/./mqtt_gateway.py", line 712, in <module>
saic-mqtt-gateway  |     asyncio.run(mqtt_gateway.run(), debug=debug_log_enabled())
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/asyncio/runners.py", line 190, in run
saic-mqtt-gateway  |     return runner.run(main)
saic-mqtt-gateway  |            ^^^^^^^^^^^^^^^^
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/asyncio/runners.py", line 118, in run
saic-mqtt-gateway  |     return self._loop.run_until_complete(task)
saic-mqtt-gateway  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
saic-mqtt-gateway  |     return future.result()
saic-mqtt-gateway  |            ^^^^^^^^^^^^^^^
saic-mqtt-gateway  |   File "/usr/src/app/./mqtt_gateway.py", line 301, in run
saic-mqtt-gateway  |     login_response_message = self.saic_api.login()
saic-mqtt-gateway  |                              ^^^^^^^^^^^^^^^^^^^^^
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/site-packages/saic_ismart_client/saic_api.py", line 119, in login
saic-mqtt-gateway  |     self.message_v1_1_coder.decode_response(login_response_hex, login_response_message)
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/site-packages/saic_ismart_client/ota_v1_1/Message.py", line 13, in decode_response
saic-mqtt-gateway  |     super().decode_response(message, decoded_message)
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/site-packages/saic_ismart_client/common_model.py", line 428, in decode_response
saic-mqtt-gateway  |     application_data_dict = self.asn1_tool_uper.decode(decoded_message.application_data.asn_type,
saic-mqtt-gateway  |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/site-packages/asn1tools/compiler.py", line 167, in decode
saic-mqtt-gateway  |     decoded = type_.decode(data)
saic-mqtt-gateway  |               ^^^^^^^^^^^^^^^^^^
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/site-packages/asn1tools/codecs/uper.py", line 500, in decode
saic-mqtt-gateway  |     raise e
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/site-packages/asn1tools/codecs/uper.py", line 496, in decode
saic-mqtt-gateway  |     return self._type.decode(decoder)
saic-mqtt-gateway  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/site-packages/asn1tools/codecs/per.py", line 830, in decode
saic-mqtt-gateway  |     decoded = self.decode_root(decoder)
saic-mqtt-gateway  |               ^^^^^^^^^^^^^^^^^^^^^^^^^
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/site-packages/asn1tools/codecs/per.py", line 848, in decode_root
saic-mqtt-gateway  |     raise e
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/site-packages/asn1tools/codecs/per.py", line 844, in decode_root
saic-mqtt-gateway  |     value = member.decode(decoder)
saic-mqtt-gateway  |             ^^^^^^^^^^^^^^^^^^^^^^
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/site-packages/asn1tools/codecs/per.py", line 1419, in decode
saic-mqtt-gateway  |     return self.decode_root(decoder)
saic-mqtt-gateway  |            ^^^^^^^^^^^^^^^^^^^^^^^^^
saic-mqtt-gateway  |   File "/usr/local/lib/python3.11/site-packages/asn1tools/codecs/per.py", line 1439, in decode_root
saic-mqtt-gateway  |     raise DecodeError(
saic-mqtt-gateway  | asn1tools.codecs.DecodeError: MPUserLoggingInResp.languageType: Expected enumeration index 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 or 16, but got 17.
saic-mqtt-gateway exited with code 1
```

My compose.yml

```
version: "3.1"  
  
services:  
   image: "saicismartapi/saic-python-mqtt-gateway"  
   container_name: "saic-mqtt-gateway"  
   environment:  
     - MQTT_URI=tcp://mqtt-server-ip:1883  
     - MQTT_USER=mqtt-username
     - MQTT_PASSWORD=mqtt-password
     - SAIC_USER=saic-username
     - SAIC_PASSWORD=saic-password
     - LOG_LEVEL=DEBUG  
     - MQTT_LOG_LEVEL=DEBUG
```

**Comment by tosate** [comment]
Hi @Xeudoron 

the grammar in `src/saic_ismart_client/ASN.1_schema/v1_1/ApplicationData.asn1`defines the following language type enumeration:
```
LanguageType ::= ENUMERATED
{
  thai(0),
  english(1),
  china(2),
  de(3),
  fr(4),
  es(5),
  nl(6),
  nn(7),
  da(8),
  sv(9),
  it(10),
  pt(11),
  pl(12),
  el(13),
  fi(14),
  arabic(15),
  hindi(16)
}
```

Obviously, the API returns 17 in your case.
`Expected enumeration index 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 or 16, but got 17.`

Which language did you setup in the App? Try to switch to English and check whether this fixes the issue!

**Comment by Xeudoron** [comment]
Hello tosate,

and thank you for your response. There are two mobile phones connected to the vehicle cloud. One is set in English and another was set in Czech language (probably 17?). It looks like that depending which connected last, the specific language is set on the server. 

The problem occurred, when we integrated our EV charger into HA and for debugging we used the phone app with Czech language which must have broke the container. After connecting with the app in English, I was able to start the addon without any issues.

For the future, all devices will be set in English language.

I sincerely appreciate your guidance.

Given that this issue may arise in the future with others, how can I, lacking Python skills, assist in mitigating such problem? Otherwise I'd at least add this problem into https://github.com/SAIC-iSmart-API/saic-python-mqtt-gateway#prerequisites.
